web-jehlomat/src/Components/Modal/StyledModal.tsx

```diff
@@ -1,79 +1,88 @@
-import { Modal } from '@mui/material';
-import TitleBar from '../../Components/Navigation/TitleBar';
+import { Dialog } from '@mui/material';
+import ModalTitleBar from '../../Components/Navigation/ModalTitleBar';
 import ChevronLeft from '@mui/icons-material/ChevronLeft';
 import CloseIcon from '@mui/icons-material/Close';
+import { useHistory } from 'react-router-dom';
 
 interface Props {
-    buttonText: string;
+    openModal: boolean;
+    toggleModal: () => void;
     modalHeaderText?: string;
 }
 
 const ModalWrapper = styled.div<{ mobile?: boolean }>`
-    background: ${white};
-    height: 100%;
-    display: flex;
-    flex-direction: column;
+    position: static;
+    width: 600px;
+    height: 530px;
+    overflow: hidden;
     ${props =>
-        !props.mobile &&
+        props.mobile &&
         `
-        width: 600px;
-        height: 531px;
-        border-radius: 10px;
-        overflow: hidden;
+        background: ${white};
+        height: 100%;
+        width: 100%;
+        display: flex;
+        position: absolute;
+        flex-direction: column;
     `}
 `;
 
 const ModalBody = styled.div<{ mobile?: boolean }>`
     padding-top: 50px;
+    height: 100%;
 `;
 
-const StyledModal: FC<Props> = ({ children, buttonText, modalHeaderText }) => {
+const StyledModal: FC<Props> = ({ children, modalHeaderText, openModal, toggleModal }) => {
     const isMobile = useMediaQuery(media.lte('mobile'));
-
-    const [openModal, setOpenModal] = useState(false);
+    let history = useHistory();
 
     const handleClose = () => {
-        setOpenModal(false);
+        toggleModal();
+        history.goBack();
     };
 
     return (
         <>
             {isMobile ? 
-                <AddButton style={{ marginLeft: '10px' }} onClick={() => setOpenModal(true)} />
-                :
-                <TextButton text={buttonText} style={{ marginLeft: '10px', color: `${primaryDark}` }} onClick={() => setOpenModal(true)} />
-            }
-
-            <Modal
-                open={openModal}
-                onClose={handleClose}
-                aria-labelledby="modal-modal-title"
-                aria-describedby="modal-modal-description"
-                style={{display:'flex', flexDirection: 'column', alignItems:'center', justifyContent:'center', background: `${white}`}}
-            >
-                <ModalWrapper mobile={isMobile}>
-                    <TitleBar
-                        icon={isMobile ? <ChevronLeft sx={{ color: white, fontSize: 40 }} /> : <CloseIcon sx={{ color: white, fontSize: 40 }} />}
-                        onIconClick={() => {
-                            setOpenModal(false);
-                        }}
+                <ModalWrapper mobile>
+                    <ModalTitleBar
+                        icon={<ChevronLeft sx={{ color: white, fontSize: 40 }} />}
+                        mobile
+                        onIconClick={() => {handleClose()}}
```

Comment:
```suggestion
                        onIconClick={handleClose}
```
The curly brackets can be omitted and also the wrapping arrow function, so you can put there the function reference (not the result of function invocation).